core/src/com/fallenflame/game/LevelController.java

```diff
@@ -1,5 +1,498 @@
 package com.fallenflame.game;
 
+import com.badlogic.gdx.math.*;
+import com.badlogic.gdx.physics.box2d.*;
+import com.badlogic.gdx.utils.*;
+import com.fallenflame.game.physics.obstacle.Obstacle;
+import com.fallenflame.game.physics.obstacle.ObstacleCanvas;
+
+import java.util.*;
+
+/** Credit to Walker White for some code reused from B2LightsDemo */
 public class LevelController {
-    //TODO
+    //  MAY NEED THESE:
+//    /** Number of velocity iterations for the constrain solvers */
+//    public static final int WORLD_VELOC = 6;
+//    /** Number of position iterations for the constrain solvers */
+//    public static final int WORLD_POSIT = 2;
+
+    // Physics objects for the level
+    /** Reference to the player character */
+    private PlayerModel player;
+    /** Reference to the exit (for collision detection) */
+    private ExitModel exit;
+    /** Reference to all enemies */
+    private List<EnemyModel> enemies;
+    /** Reference to all walls */
+    private List<WallModel> walls;
+    /** Reference to all flares */
+    private List<FlareModel> flares;
+    /** Level Model for AI Pathfinding */
+    private LevelModel levelModel;
+    /** Flare JSONValue */
+    private JsonValue flareJSON;
+
+    /** Whether or not the level is in debug more (showing off physics) */
+    private boolean debug;
+
+    // World Definitions
+    /** The Box2D world */
+    protected World world;
+    /** The boundary of the world */
+    protected Rectangle bounds;
+    /** The world scale */
+    protected Vector2 scale;
+
+    // Controllers
+    /** Light Controller */
+    private LightController lightController;
+    /** AI Controllers */
+    private List<AIController> AIControllers;
+
+    /** Enum to specify level state */
+    public enum LevelState {
+        /** Player has reached the exit */
+        WIN,
+        /** Player has died */
+        LOSS,
+        /** Player is still playing */
+        IN_PROGRESS
+    }
+    private LevelState levelState;
+
+    // TO FIX THE TIMESTEP
+    /** The maximum frames per second setting for this level */
+    protected int maxFPS;
+    /** The minimum frames per second setting for this level */
+    protected int minFPS;
+    /** The amount of time in to cover a single animation frame */
+    protected float timeStep;
+    /** The maximum number of steps allowed before moving physics forward */
+    protected float maxSteps;
+    /** The maximum amount of time allowed in a frame */
+    protected float maxTimePerFrame;
+    /** The amount of time that has passed without updating the frame */
+    protected float physicsTimeLeft;
+
+    /**
+     * Returns the bounding rectangle for the physics world
+     *
+     * The size of the rectangle is in physics, coordinates, not screen coordinates
+     *
+     * @return the bounding rectangle for the physics world
+     */
+    public Rectangle getBounds() { return bounds; }
```

Comment:
Not absolutely sure here, but I think we'll need a setBounds too if we want a resizeable canvas (or any sized canvas that isn't set in the level JSON